components/sup/src/manager/mod.rs

```diff
@@ -270,22 +270,20 @@ impl Manager {
         opts: protocol::ctl::SvcStatus,
     ) -> NetResult<()> {
         let statuses = Self::status(&mgr.cfg)?;
-        if statuses.is_empty() {
-            req.reply_complete(net::ok());
-            return Ok(());
-        }
         if let Some(ident) = opts.ident {
             for status in statuses {
                 if status.pkg.ident.satisfies(&ident) {
                     let mut msg: protocol::types::ServiceStatus = status.into();
                     req.reply_complete(msg);
-                    break;
+                    return Ok(());
```

Comment:
Without this `return`, the case of a service whose status is present in `statuses` returned `Err`, but that was masked by the call to `reply_complete`.